doc/neps/nep-0056-array-api-main-namespace.rst

```diff
@@ -0,0 +1,602 @@
+the array API standard chose to follow C99 and other libraries in using
+``acos``, ``asin`` etc. rather than ``arccos``, ``arcsin``, etc. NumPy usually
+also follows C99; it is not entirely clear why this naming choice was made many
+years ago.
+
+In total 13 aliases are added to the main namespace and 2 aliases to
+``numpy.linalg``:
+
+- trigonometry functions: ``acos``, ``acosh``, ``asin``, ``asinh``, ``atan``,
+  ``atanh``, ``atan2``
+- bit-wise functions: ``bitwise_left_shift``, ``bitwise_invert``,
+  ``bitwise_right_shift``
+- other functions: ``concat``, ``permute_dims``, ``pow``
+- in ``numpy.linalg``: ``tensordot``, ``matmul``
+
+In the future NumPy can choose to hide the original names from its ``__dir__``
+to nudge users to the preferred spelling for each function.
+
+
+New keywords with overlapping semantics
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Similarly to function name aliases, there are a couple of new keywords which
+have overlap with existing ones:
+
+- ``correction`` keyword for ``std`` and ``var`` (overlaps with ``ddof``)
+- ``stable`` keyword for ``sort`` and ``argsort`` (overlaps with ``kind``)
+
+The ``correction`` name is for clarity ("delta degrees of freedom" is not easy
+to understand). ``stable`` is complementary to ``kind``, which already has
+``'stable'`` as an option (a separate keyword may be more discoverable though
+and hence nice to have anyway), allowing a library to reserve the right to
+change/improve the stable and unstable sorting algorithms.
+
+
+New ``unique_*`` functions
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``unique`` function, with ``return_index``, ``return_inverse``, and
+``return_counts`` arguments that influence the cardinality of the returned
+tuple, is replaced in the array API by four respective functions:
+``unique_all``, ``unique_counts``, ``unique_inverse``, and ``unique_values``.
+These new functions avoid polymorphism, which tends to be a problem for JIT
+compilers and static typing. Use of these functions therefore helps tools like
+Numba as well as users of static type checkers like Mypy.
+
+
+``np.bool`` addition
+^^^^^^^^^^^^^^^^^^^^
+
+One of the aliases that used to live in NumPy but was removed is ``np.bool``.
+To comply with the array API it was reintroduced with a different meaning, as
+now it points to NumPy's bool instead of a Python builtin. This change is a
+good idea and we were planning to make it anyway, because ``bool`` is a nicer
+name than ``bool_``. However, we may not have scheduled that reintroduction of
+the name for 2.0 if it had not been part of the array API standard.
+
+
+Parts of the standard that are not adopted
+------------------------------------------
+
+There are a couple of things that the standard prescribes which we propose *not*
+to follow (at least at this time). These are:
+
+1. The requirement that an indexing operation must result in an array of the
+   same dtype of the indexed array.
```

Comment:
Thanks for this section, but not sure where this point comes from?  Sure it returns a different type, but not dtype.
(You "lose" the byte-order storage detail and metadata, but I am sure you don't care about either.)

You can mention that the array-api prefers to not have scalars in some form I guess, for the second part, maybe for that it makes sense to use indexing as an example (it isn't the only place, especially right now).